protocol/communities/community_test.go

```diff
@@ -143,6 +143,60 @@ func (s *CommunitySuite) TestCreateChat() {
 	s.Require().NotNil(changes.ChatsAdded[newChatID])
 }
 
+func (s *CommunitySuite) TestEditChat() {
+	newChatID := "new-chat-id"
+	org := s.buildCommunity(&s.identity.PublicKey)
+
+	identity := &protobuf.ChatIdentity{
+		DisplayName: "new-chat-display-name",
+		Description: "new-chat-description",
+	}
+	permissions := &protobuf.CommunityPermissions{
+		Access: protobuf.CommunityPermissions_NO_MEMBERSHIP,
+		Private: false,
+	}
+
+	_, err := org.CreateChat(newChatID, &protobuf.CommunityChat{
+		Identity:    identity,
+		Permissions: permissions,
+	})
+	s.Require().NoError(err)
+
+	org.config.PrivateKey = nil
+	editedIdentity := &protobuf.ChatIdentity{
+		DisplayName: "edited-new-chat-display-name",
+		Description: "edited-new-chat-description",
+	}
+	editedPermissions := &protobuf.CommunityPermissions{
+		Access: protobuf.CommunityPermissions_NO_MEMBERSHIP,
+		Private: true,
+	}
+	_, err = org.EditChat(newChatID, &protobuf.CommunityChat{
+		Identity:    editedIdentity,
+		Permissions: editedPermissions,
+	})
+	s.Require().Equal(ErrNotAdmin, err)
+	
+	description := org.config.CommunityDescription
+	org.config.PrivateKey = s.identity
+	editChanges, err := org.EditChat(newChatID, &protobuf.CommunityChat{
+		Identity:    editedIdentity,
+		Permissions: editedPermissions,
+	})
+
+	s.Require().NoError(err)
```

Comment:
Test should be updated to include work with database and ensuring chat description retainted after saving/loading